examples/test_cpp/test_data_provider.cc

```diff
@@ -80,12 +94,39 @@ std::string test_data_provider::create_json_outcome(size_t thread_id, size_t exa
   return oss.str();
 }
 
-std::string test_data_provider::create_context_json(const std::string& cntxt, const std::string& action) const {
+std::string test_data_provider::create_context_json(const std::string& cntxt, const std::string& action, bool ccb) const {
+  std::ostringstream oss;
+  if (!ccb)
+    oss << "{ " << cntxt << ", " << action << " }";
+  else
+    oss << cntxt << ", " << action;
+
+  return oss.str();
+}
+
+std::string test_data_provider::create_ccb_context_json(const std::string& cntxt, const std::vector<std::string>& ids) const {
   std::ostringstream oss;
-  oss << "{ " << cntxt << ", " << action << " }";
+  oss << "{ " << cntxt << ", " << create_slots_json(ids) << " }";
   return oss.str();
 }
 
+std::string test_data_provider::create_slots_json(const std::vector<std::string>& ids) const {
```

Comment:
Aside: I really really wish we could use fmtlib...

Reply:
agree